src/main/java/appeng/me/storage/ItemCellInventory.java

```diff
@@ -0,0 +1,316 @@
+		{
+			return true;
+		}
+		return BLACK_LIST.contains( ( input.getItemDamage() << Platform.DEF_OFFSET ) | Item.getIdFromItem( input.getItem() ) );
+	}
+
+	@Override
+	public IAEItemStack injectItems( final IAEItemStack input, final Actionable mode, final IActionSource src )
+	{
+		if( input == null )
+		{
+			return null;
+		}
+		if( input.getStackSize() == 0 )
+		{
+			return null;
+		}
+
+		if( isBlackListed( input ) || this.cellType.isBlackListed( this.i, input ) )
+		{
+			return input;
+		}
+
+		final ItemStack sharedItemStack = input.createItemStack();
+
+		if( ItemCellInventory.isStorageCell( sharedItemStack ) )
+		{
+			final IMEInventory meInventory = getCell( sharedItemStack, null );
+			if( meInventory != null && !this.isEmpty( meInventory ) )
+			{
+				return input;
+			}
+		}
+
+		final IAEItemStack l = this.getCellItems().findPrecise( input );
+		if( l != null )
+		{
+			final long remainingItemSlots = this.getRemainingItemCount();
+			if( remainingItemSlots < 0 )
+			{
+				return input;
+			}
+
+			if( input.getStackSize() > remainingItemSlots )
+			{
+				final IAEItemStack r = input.copy();
+				r.setStackSize( r.getStackSize() - remainingItemSlots );
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() + remainingItemSlots );
+					this.updateItemCount( remainingItemSlots );
+					this.saveChanges();
+				}
+				return r;
+			}
+			else
+			{
+				if( mode == Actionable.MODULATE )
+				{
+					l.setStackSize( l.getStackSize() + input.getStackSize() );
+					this.updateItemCount( input.getStackSize() );
+					this.saveChanges();
+				}
+				return null;
+			}
+		}
+
+		if( this.canHoldNewItem() ) // room for new type, and for at least one item!
+		{
+			final int remainingItemCount = (int) this.getRemainingItemCount() - this.getBytesPerType() * 8;
+			if( remainingItemCount > 0 )
+			{
+				if( input.getStackSize() > remainingItemCount )
+				{
+					final ItemStack toReturn = sharedItemStack.copy();
+					toReturn.setCount( sharedItemStack.getCount() - remainingItemCount );
+					if( mode == Actionable.MODULATE )
+					{
+						final ItemStack toWrite = sharedItemStack.copy();
+						toWrite.setCount( remainingItemCount );
+
+						this.cellItems.add( AEItemStack.fromItemStack( toWrite ) );
+						this.updateItemCount( toWrite.getCount() );
+
+						this.saveChanges();
+					}
+					return AEItemStack.fromItemStack( toReturn );
+				}
+
+				if( mode == Actionable.MODULATE )
+				{
+					this.updateItemCount( input.getStackSize() );
+					this.cellItems.add( input );
+					this.saveChanges();
+				}
+
+				return null;
+			}
+		}
+
+		return input;
+	}
+
+	@Override
+	public IAEItemStack extractItems( final IAEItemStack request, final Actionable mode, final IActionSource src )
+	{
+		if( request == null )
+		{
+			return null;
+		}
+
+		final long size = Math.min( Integer.MAX_VALUE, request.getStackSize() );
+
+		IAEItemStack Results = null;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00117)